packages/agent/src/utils/callAndPoll.ts

```diff
@@ -0,0 +1,54 @@
+import { Principal } from '@dfinity/principal';
+import { Agent, Certificate, bufFromBufLike, polling, v3ResponseBody } from '..';
+import { AgentError } from '../errors';
+
+/**
+ * Call a canister using the v3 api and either return the response or fall back to polling
+ * @param options - The options to use when calling the canister
+ * @param options.canisterId - The canister id to call
+ * @param options.methodName - The method name to call
+ * @param options.agent - The agent to use to make the call
+ * @param options.arg - The argument to pass to the canister
+ * @returns The certificate response from the canister (which includes the reply)
+ */
+export async function callAndPoll(options: {
+  canisterId: Principal | string;
+  methodName: string;
+  agent: Agent;
+  arg: ArrayBuffer;
+}): Promise<ArrayBuffer> {
```

Comment:
We need to get back the `contentMap` _and_ the `rawCert`. Ideally, this function would have the exact same interface as `call`, maybe with the addition of being able to supply the `pollStrategy` as suggested by @dfx-json.

Also, please extend the `CallOptions` interface to allow specifying the `nonce` too.

Reply:
@frederikrothenberger can you define what `contentMap` precisely is? Should it be an ArrayBuffer? Does it need to be decoded into a JavaScript object? Are you willing to provide an IDL interface for the types so we can actually decode the candid?

Relatedly, I don't know what actual feature this is for, or how it will be used in practice. We'd probably have had a lot less back-and-forth about this if I was just designing a feature for your use-case instead of all these half-requirements and us each having slightly different definitions of what things are